esp/smc/SMCLib/InfoCacheReader.cpp

```diff
@@ -0,0 +1,77 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2019 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#pragma warning (disable : 4786)
+#include "InfoCacheReader.hpp"
+
+bool CInfoCache::isCachedInfoValid(unsigned timeOutSeconds)
+{
+    CDateTime timeNow;
+    timeNow.setNow();
+    return timeNow.getSimple() <= timeCached.getSimple() + timeOutSeconds;;
+}
+
+void CInfoCacheReaderThread::threadmain()
+{
+#ifndef _CONTAINERIZED
```

Comment:
why disabled ?
What happens to the activity page with this not running?

If you don't want it running, would be better not to start the thread.

Reply:
This is copied from https://github.com/hpcc-systems/HPCC-Platform/commit/9ed3df170d61903ec4f9b3494eb83a5a6944a548. To my understand, the code is disabled for now and we may add some code later for CONTAINERIZED using else. For now, the activity page will be blank for CONTAINERIZED.

Reply:
ok not new to this PR then.

Reply:
@wangkx - the activity page in CONTAINERIZED mode needs revisiting anyway, but it's beyond the scope of this PR.

Your last commit added various #ifdef _CONTAINERIZED's - I think it would be better to leave the one that was already there and revert the others you added.